bootloader/src/pyi_archive.c

```diff
@@ -398,6 +538,8 @@ pyi_arch_open(ARCHIVE_STATUS *status)
      */
 #if defined(WIN32) || defined(__APPLE__)
     search_end = findDigitalSignature(status);
+#else
+    search_end = findElfSection(status);
```

Comment:
Similar as at the top of the file: we assume that all platforms except Windows and macOS use ELF and work the same way as Linux. I think it's safer to have a check just for Linux and call the function only there.

Reply:
See above. Is it incorrect to assume ELF is being used on the other Unixes?

Note: If the pydata section is not found by the function call, it falls back to the search from end of file here.